addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java

```diff
@@ -0,0 +1,280 @@
+	            return true;
+	        case PARAM_URL:
+	            url = value;
+	            return true;
+            case PARAM_FAIL_IF_LESS_URLS:
+                failIfFoundUrlsLessThan = Integer.parseInt(value);
+                return true;
+            case PARAM_WARN_IF_LESS_URLS:
+                warnIfFoundUrlsLessThan = Integer.parseInt(value);
+                return true;
+            case PARAM_MAX_DURATION:
+                maxDuration = Integer.parseInt(value);
+                // Don't consume this as we still want it to be applied to the spider params
+                return false;
+            default:
+                // Ignore
+                break;
+        }
+        return false;
+    }
+
+    @Override
+    public Map<String, String> getCustomConfigParameters() {
+        Map<String, String> map = super.getCustomConfigParameters();
+        map.put(PARAM_CONTEXT, "");
+        map.put(PARAM_URL, "");
+        map.put(PARAM_FAIL_IF_LESS_URLS, "0");
+        map.put(PARAM_WARN_IF_LESS_URLS, "0");
+        return map;
+    }
+
+	@Override
+	public void runJob(AutomationEnvironment env, LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+
+        Context context;
+        if (contextName != null) {
+            context = env.getContext(contextName);
+            if (context == null) {
+                progress.error(
+                        Constant.messages.getString(
+                                "automation.error.context.unknown",
+                                env.getUrlStringForContext(context)));
+                return;
+            }
+        } else {
+            context = env.getDefaultContext();
+        }
+
+        URI uri = null;
+        try {
+            if (url != null) {
+                uri = new URI(url);
+            } else {
+                uri = new URI(env.getUrlStringForContext(context).toString());
+            }
+        } catch (Exception e1) {
+            progress.error(
+                    Constant.messages.getString(
+                            "automation.error.context.badurl",
+                            env.getUrlStringForContext(context)));
+            return;
+        }
+
+        AjaxSpiderTarget.Builder targetBuilder =
+                AjaxSpiderTarget.newBuilder(Model.getSingleton().getSession())
+                		.setContext(context)
+                        .setInScopeOnly(inScopeOnly)
+                        .setOptions(getExtSpider().getAjaxSpiderParam())
+                        .setStartUri(uri)
+                        .setSubtreeOnly(false);
+
+        AjaxSpiderTarget target = targetBuilder.build();
+		JobSpiderListener listener = new JobSpiderListener();
+
+		getExtSpider().startScan(uri.toString(), target, listener);
+
+        long endTime = Long.MAX_VALUE;
+        if (maxDuration > 0) {
+            // The spider should stop, if it doesnt we will stop it
+            endTime = System.currentTimeMillis() + (maxDuration * 1000 * 10) + 20;
```

Comment:
Use TimeUnit to make this a bit more readable?

Reply:
Arg, should have remembered that - copied the spider code before making that fix ;)